ring/partition_ring_test.go

```diff
@@ -966,6 +966,137 @@ func TestPartitionRing_ShuffleShardWithLookback_CachingConcurrency(t *testing.T)
 	assert.Empty(t, ring.shuffleShardCache.cacheWithoutLookback)
 }
 
+func TestPartitionRingGetTokenRangesForPartition(t *testing.T) {
```

Comment:
To further assert correctness in a more extensive way, you could consider another tests where you create a ring with many active partitions, then you grab the list of tokens for all partitions, and for each of T-1, T, T+1 you ensure `ActivePartitionForKey()` matches with the token ranges owned by that partition.

Reply:
Done in [`7f9f1bf` (#488)](https://github.com/grafana/dskit/pull/488/commits/7f9f1bf7b6b2b0ad4599d23690fc88ad66db4ec0).